### PR TITLE
SONARJAVA-1047 RSPEC-2676 Neither "Math.abs" nor negation should be used on numbers that could be "MIN_VALUE"

### PR DESCRIPTION
**Pull request by merciesa**

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

File: java-checks/src/main/java/org/sonar/java/checks/AbsOnNegativeCheck.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      }
+    } else {
+      ExpressionTree operand = ((UnaryExpressionTree) tree).expression();
+      checkForIssue(operand);
+    }
+  }
+
+  private void checkForIssue(ExpressionTree tree) {
+    MethodInvocationTree nestedTree = extractMethodInvocation(tree);
+    if (nestedTree != null && NEGATIVE_METHODS.anyMatch(nestedTree)) {
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/AbsOnNegativeCheck.java]: And what about testing if the expression is directly Integer.MIN_VALUE ?

**Comment by SonarTech** [comment]
SonarQube analysis reported no new issues.

**Comment by merciesa** [comment]
I won't add support for explicit values now, because LiteralUtils fails for the two specific values.

**Comment by m-g-sonar** [comment]
LGTM!

**Pull request closed without merging**